src/pyfme/models/systems.py

```diff
@@ -141,153 +197,360 @@ def q(self):
+    def time(self):
+        return self.model.time
+
+    @time.setter
+    def time(self, value):
+        self.model.time = value
+
+    def set_initial_state(self,
+                          geodetic_coord=None, geocentric_coord=None,
+                          euler_angles=None,
+                          quaternions=None,
+                          vel_body=None, vel_NED=None,
+                          vel_ang=None,
+                          accel_body=None, accel_NED=None,
+                          accel_ang=None):
+        """Initialize the system state and the model state.
 
         Parameters
         ----------
-        lat, lon, h: float
-            Latitude, longitude and height (rad, rad, m).
-        psi, x_earth, y_earth: float, opt
-            Yaw angle and initial Earth position (rad, m, m).
-        integrator: str, optional
-            Any allowed integrator for `ode` class: "vode", "zvode", "lsoda",
-            "dopri5", "dop853".
-        use_jac : bool, optional
-            Use analytical jacobians of system equations.
+        geodetic_coord : ndarray, shape(3), opt
+            (lat [rad], lon [rad], height [m]). Only if geodetic_coord are
+            not given.
+        geocentric_coord : ndarray, shape(3), opt
+            (x_geo [m], y_geo [m], z_geo [m]). Only if geocentric_coord are
+            not given.
+        euler_angles : ndarray, shape(3), opt
+            (psi [rad], theta [rad], phi [rad]). Only if quaternions are not
+            given.
+        quaternions : ndarray, shape(4), opt
+            (q0, q1, q2, q3). Only if euler_angles are not given.
+        vel_body : ndarray, shape(3), opt
+            (u [m/s], v [m/s], w [m/s]). Only if vel_NED is not given.
+        vel_NED : ndarray, shape(3), opt
+            (v_north [m/s], v_east [m/s], v_down [m/s]). Only if vel_body is
+            not given.
+        vel_ang : ndarray, shape(3), opt
+            (p [rad/s], q [rad/s], r [rad/s])
+        accel_body : ndarray, shape(3), opt
+            (u_dot [m/s²], v_dot [m/s²], w_dot [m/s²]). Only if accel_NED is
+            not given.
+        accel_NED : ndarray, shape(3), opt
+            (VN_dot [m/s²], VE_dot [m/s²], VD_dot [m/s²]). Only if
+            accel_body is not given.
+        accel_ang : ndarray, shape(3), opt
+            (p_dot [rad/s²], q_dot [rad/s²], r_dot [rad/s²])
         """
-        super().__init__(lat, lon, h, psi, x_earth, y_earth)
-        # State vector must be initialized with set_initial_state_vector() method
-        self.state_vector = None
-
-        from pyfme.models.euler_flat_earth import lamceq, kaeq, kleq
-        self.lamceq = lamceq
-
-        if use_jac:
-            from pyfme.models.euler_flat_earth import lamceq_jac, kaeq_jac
-            jac_LM_and_AM = lamceq_jac
-            jac_att = kaeq_jac
-            jac_nav = None  # not implemented
+
+        # Set position
+        if (geodetic_coord is not None) and (geocentric_coord is not None):
```

Comment:
The overall cyclomatic complexity of this method is probably very big. Any chance to slice it into smaller ones?